src/service/vulnerabilities/rag_poisoning_lab.py

```
@@ -0,0 +1,225 @@
+
+    assistant_output = "Code generated successfully using retrieved context."
+
+    # Actually call the LLM with poisoned context
+    try:
+        generated_code = await chat_completion(
+            system_prompt=system_prompt,
+            user_input=effective_user_input,
+            model=selected_model,
+            temperature=selected_temperature,
+        )
+    except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)